hazelcast/src/main/java/com/hazelcast/nio/VersionSettable.java

```diff
@@ -0,0 +1,37 @@
+/*
+ * Copyright (c) 2008-2017, Hazelcast, Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.hazelcast.nio;
+
+import com.hazelcast.version.Version;
+
+/**
+ * Enables setting the version on the implementing object.
+ * It may be any Version (cluster version, node version, custom version, etc.) It's up to the implementer.
+ *
+ * @since 3.9
+ */
+
+public interface VersionSettable {
```

Comment:
I think it is better to introduce `VersionSettable` in a separate PR to make change more visible or maybe in a separate commit, now i'm questioning the relation of it with this PR. Or maybe i'm still missing something